server/src/main/java/com/epam/aidial/core/server/controller/RouteController.java

```diff
@@ -161,8 +161,6 @@ private void handleProxyResponse(HttpClientResponse proxyResponse) {
             context.getUpstreamRoute().succeed();
             proxy.getRateLimiter().increase(context, context.getRoute()).onFailure(error -> log.warn("Failed to increase limit. Trace: {}. Span: {}",
                     context.getTraceId(), context.getSpanId(), error));
-        } else {
-            context.getUpstreamRoute().fail(proxyResponse);
```

Comment:
It looks like this `else` can be left. I added a test

```
   @Test
    void route_404() {
        TestWebServer.Handler handler = request -> new MockResponse().setResponseCode(404);
        try (TestWebServer server = new TestWebServer(9876, handler)) {
            Response resp = send(HttpMethod.GET, "/v1/plain", null, null, "api-key", "vstore_user_key");
            assertEquals(404, resp.status());
        }
    }
```

to `com.epam.aidial.core.server.RouteApiTest`. 

For me, the controller works fine with `context.getUpstreamRoute().fail(proxyResponse);`

Reply:
could you pls add a test case to this pr

Reply:
fixed